Option to disable aggregates by default and for specific tables (#49)
graphile/pg-aggregates: README.md

See [src/AggregateSpecsPlugin.ts](src/AggregateSpecsPlugin.ts) for examples and
more information.

## Disable aggregates

By default, aggregates are created for all tables. This significantly increases the size of your
GraphQL schema, and could also be a security (DoS) concern as aggregates can be expensive. We
recommend that you use the `-aggregates` default behavior to disable aggregates by
default, and then enable them only for the tables you need:

```ts
// graphile.config.mjs

export default {
  // ...
  schema: {
    defaultBehavior: "-aggregates",
  },
};
```

Enable aggregates for a specific table:

```sql
COMMENT ON TABLE my_schema.my_table IS E'@behavior +aggregates';
```

You also can keep aggregates enabled by default, but disable aggregates for specific tables:

```sql
COMMENT ON TABLE my_schema.my_table IS E'@behavior -aggregates';
```

> You can continue to use your `postgraphile.tags.json5` file if you used it for postgraphile V4

Reviewer: ```suggestion
You can also use a smart tags file such as `postgraphile.tags.json5` to add these behaviors:
```

>  - to enable aggregates for a specific table

Reviewer: ```suggestion
- to enable aggregates for a specific table:
```

```json
"class": {
  "my_schema.my_table": {
    "tags": {

>       "aggregates": "on"

Reviewer: ```suggestion
      "behavior": "+aggregates"
```

    }
  }
}
```

>  - to disable aggregates for a specific table

Reviewer: ```suggestion
- to disable aggregates for a specific table:
```

```json
"class": {
  "my_schema.my_table": {
    "tags": {

>       "aggregates": "off"

Reviewer: ```suggestion
      "behavior": "-aggregates"
```

    }
  }
}
```

> You can continue to use yor `@aggregates` smart tags if you already have them - it isn't necessary to replace them to behaviors.

Reviewer: ```suggestion
You can continue to use yor `@aggregates` smart tags and smart comments if you already have them - it isn't necessary to replace them to behaviors.
```

See [src/AggregateSpecsPlugin.ts](src/AggregateSpecsPlugin.ts) for examples and
more information.

## Disable aggregates

By default, aggregates are created for all tables. This significantly increases the size of your
GraphQL schema, and could also be a security (DoS) concern as aggregates can be expensive. We
recommend that you use the `-aggregates` default behavior to disable aggregates by
default, and then enable them only for the tables you need:

> (This currently doesn't work due to a [bug in PostGraphile that's being worked on](https://github.com/graphile/crystal/pull/1803))

Reviewer: ```suggestion
(This currently doesn't work due to a [bug in PostGraphile that's being worked on](https://github.com/graphile/crystal/pull/1803).)
```

};
```

The `aggregates:filterBy` behavior is used to enable/disable the [filtering by aggregates](#filtering-by-aggregates)

The `aggregates:orderBy` behavior is used to enable/disable the [ordering by aggregates](#ordering-by-aggregates)

You can use any combination of `aggregates`, `aggregates:filterBy` and `aggregates:orderBy` behaviors. 

Enable aggregates for a specific table:

```sql
COMMENT ON TABLE my_schema.my_table IS E'@behavior +aggregates +aggregates:filterBy +aggregates:orderBy';
```

You also can keep aggregates enabled by default, but disable aggregates for specific tables:

```sql
COMMENT ON TABLE my_schema.my_table IS E'@behavior -aggregates -aggregates:filterBy -aggregates:orderBy';
```

You can also use a smart tags file such as `postgraphile.tags.json5` to add these behaviors:

- to enable aggregates for a specific table:
```json
"class": {
  "my_schema.my_table": {
    "tags": {
      "behavior": "+aggregates +aggregates:filterBy +aggregates:orderBy"
    }
  }
}
```

- to disable aggregates for a specific table:
```json
"class": {
  "my_schema.my_table": {
    "tags": {
      "behavior": "-aggregates -aggregates:filterBy -aggregates:orderBy"
    }
  }
}
```

> You can continue to use yor `@aggregates` smart tags and smart comments if you already have them - it isn't necessary to replace them to behaviors.

Reviewer: ```suggestion
You can continue to use your `@aggregates` smart tags and smart comments if you already have them - it isn't necessary to replace them with behaviors.
```
